src/routes/bounce_rules_detailed.jsx

```diff
@@ -40,111 +57,92 @@ export default class BounceRuleDetailedPage extends React.Component {
-        });
-        break;
-      }
-      default: {
-        break;
-      }
+  onChangeRuleInt(e) {
+    const re = /^[0-9\b]+$/;
+    const { currentRule } = this.state;
+    const { id, value } = e.currentTarget;
+    if (re.test(value)) {
+      this.setState({
+        currentRule: { ...currentRule, [id]: parseInt(value, 10) },
+      });
     }
   }
 
-  handleButtonClicked(event) {
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+    });
+  }
+
+  handleChangelogClicked(e) {
+    const { changelog } = this.state;
+    const changeIndex = e.currentTarget.getAttribute("index");
+    this.setState({
+      selectedChange: changelog[changeIndex],
+      isChangeModalOpen: true,
+    });
+  }
+
+  handleEditClicked(e) {
+    const { id } = e.currentTarget;
+    const { currentRule } = this.state;
+    this.setState({
+      [id]: true,
+      prevRule: { ...currentRule },
+    });
+  }
+
+  handleCancelSaveClicked(e) {
+    const { id } = e.currentTarget;
     const { currentRule, prevRule } = this.state;
-    const { id } = event.currentTarget;
-    switch (id) {
-      case "cancelClicked": {
-        if (!_.isEqual(prevRule, currentRule)) {
-          this.setState({
-            isCancelConfirmOpen: true,
-          });
-        } else {
-          this.setState({
-            isEditClicked: false,
-          });
-        }
-        break;
-      }
-      case "editClicked": {
-        this.setState({
-          isEditClicked: true,
-          prevRule: { ...currentRule },
-        });
-        break;
-      }
-      case "saveClicked": {
-        if (!_.isEqual(prevRule, currentRule)) {
-          this.setState({
-            isConfirmOpen: true,
-          });
-        } else {
-          this.setState({
-            isEditClicked: false,
-          });
-        }
-        break;
-      }
-      case "changeClicked": {
-        this.setState({
-          isChangeModalOpen: true,
-        });
-        break;
-      }
-      default: {
-        break;
-      }
+    if (!_.isEqual(prevRule, currentRule)) {
+      this.setState({
+        [id]: true,
+      });
+    } else {
+      this.setState({
+        isEditClicked: false,
+      });
     }
   }
 
-  handleModalConfirm(event) {
+  handleCancelConfirmation() {
     const { prevRule } = this.state;
-    const { id } = event.currentTarget;
-    switch (id) {
-      case "cancelModal": {
-        this.setState({
-          currentRule: { ...prevRule },
-          isCancelConfirmOpen: false,
-          isEditClicked: false,
-        });
-        break;
-      }
-      case "saveModal": {
+    this.setState({
+      currentRule: { ...prevRule },
+      isCancelConfirmOpen: false,
+      isEditClicked: false,
+    });
+  }
+
+  async handleSaveConfirmation() {
+    const { currentRule } = this.state;
+    const { id } = currentRule;
+    const newRule = _.omit(currentRule, ["created_at"]);
+    await putRule(id, newRule);
```

Comment:
What if the put fails here in this error case? Will we handle the error case for failing to update in another ticket?

Reply:
will be handled in a seperate ticket